src/test/java/com/exactpro/junit5/WithByteBufExtension.java

```diff
@@ -0,0 +1,97 @@
+
+import com.exactpro.epfast.ByteBufUtils;
+import io.netty.buffer.ByteBuf;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.TestTemplateInvocationContext;
+import org.junit.jupiter.api.extension.TestTemplateInvocationContextProvider;
+import org.junit.platform.commons.util.Preconditions;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.stream.Stream;
+
+import static org.junit.platform.commons.util.AnnotationUtils.isAnnotated;
+
+public class WithByteBufExtension implements TestTemplateInvocationContextProvider {
+
+    private static final String METHOD_CONTEXT_KEY = "context";
+
+    @Override
+    public boolean supportsTestTemplate(ExtensionContext context) {
+        if (!context.getTestMethod().isPresent()) {
+            return false;
+        }
+        Method testMethod = context.getTestMethod().get();
+        if (!isAnnotated(testMethod, WithByteBuf.class)) {
+            return false;
+        }
+        WithByteBufMethodContext methodContext = new WithByteBufMethodContext(testMethod);
+        Preconditions.condition(methodContext.hasPotentiallyValidSignature(testMethod.getParameters()),
+            () -> String.format(
+                "@WithByteBuf method [%s] declares invalid formal parameter: "
+                    + "argument count must be 1 "
+                    + "argument type must be Collection class.",
+                testMethod.toGenericString()));
+
+        getStore(context).put(METHOD_CONTEXT_KEY, methodContext);
+        return true;
+    }
+
+    @Override
+    public Stream<TestTemplateInvocationContext> provideTestTemplateInvocationContexts(
+        ExtensionContext extensionContext) {
+        Method templateMethod = extensionContext.getRequiredTestMethod();
+        String displayName = extensionContext.getDisplayName();
+        WithByteBufMethodContext methodContext = getStore(extensionContext)
+            .get(METHOD_CONTEXT_KEY, WithByteBufMethodContext.class);
+        String argValue = templateMethod.getAnnotation(WithByteBuf.class).value();
+        checkArgumentFormat(argValue);
+        return generateBufferArguments(argValue)
+            .stream()
+            .map(arguments -> createInvocationContext(
+                WithBufNameFormatter.formatName(displayName, arguments), methodContext, arguments));
+    }
+
+    private void checkArgumentFormat(String hexString) throws IllegalArgumentFormatException {
+        try {
+            ByteBufUtils.fromHex(hexString);
+        } catch (Exception e) {
+            throw new IllegalArgumentFormatException("Illegal Argument Format");
+        }
+    }
+
+    private Collection<Collection<ByteBuf>> generateBufferArguments(String hexString) {
+        Collection<ByteBuf> bufArray = new ArrayList<>();
+        Collection<Collection<ByteBuf>> argsArray = new ArrayList<>();
+
+        bufArray.add(ByteBufUtils.fromHex(hexString));
```

Comment:
Change fromHex to return `byte[]` or create new method and reuse in existing fromHex. Use this byte array as a source for your computations